invenio_oauthclient/contrib/cern.py

```diff
@@ -209,12 +209,19 @@ def get_resource(remote):
 def account_info(remote, resp):
     """Retrieve remote account information used to find local user."""
     res = get_resource(remote)
+    session.pop('cern_resource')
 
     email = res['EmailAddress'][0]
     external_id = res['PersonID'][0]
     nice = res['CommonName'][0]
     name = res['DisplayName'][0]
 
+    groups = fetch_groups(res['Group'])
+    provides = [UserNeed(email)] + \
+               [RoleNeed('{0}@cern.ch'.format(group)) for group in groups]
+
+    session['identity.cern_provides'] = provides
```

Comment:
I think one possible attack could be something along the lines of:

1) Try to login as Alice who is member of CMS.
2) Now, login with Bob which is not member of CMS via different authentication mechanism (either local login or another oauth). -> Now Bob is possibly a member of CMS.

Essentially the trick is to get the groups stored on the session, and get the ``identity_loaded`` signal fired in some different manner.

Why are you in need of the groups at this point in time? Perhaps there's a different way.